agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java

```diff
@@ -243,46 +242,43 @@ private void exportRemoteDependency(SpanData span, Map<AttributeKey<?>, Object>
+        setExtraAttributes(telemetry.getProperties(), attributes);
         track(telemetry, samplingPercentage);
         trackEvents(span, samplingPercentage);
     }
 
-    private void applySemanticConventions(Map<AttributeKey<?>, Object> attributes, RemoteDependencyTelemetry telemetry, Span.Kind spanKind) {
-        String httpMethod = removeAttributeString(attributes, SemanticAttributes.HTTP_METHOD);
+    private void applySemanticConventions(ReadableAttributes attributes, RemoteDependencyTelemetry telemetry, Span.Kind spanKind) {
+        String httpMethod = attributes.get(SemanticAttributes.HTTP_METHOD);
         if (httpMethod != null) {
             applyHttpClientSpan(attributes, telemetry);
             return;
         }
-        String rpcSystem = removeAttributeString(attributes, SemanticAttributes.RPC_SYSTEM);
+        String rpcSystem = attributes.get(SemanticAttributes.RPC_SYSTEM);
         if (rpcSystem != null) {
             applyRpcClientSpan(attributes, telemetry, rpcSystem);
             return;
         }
-        String dbSystem = removeAttributeString(attributes, SemanticAttributes.DB_SYSTEM);
+        String dbSystem = attributes.get(SemanticAttributes.DB_SYSTEM);
         if (dbSystem != null) {
             applyDatabaseClientSpan(attributes, telemetry, dbSystem);
             return;
         }
-        String messagingSystem = removeAttributeString(attributes, SemanticAttributes.MESSAGING_SYSTEM);
+        String messagingSystem = attributes.get(SemanticAttributes.MESSAGING_SYSTEM);
         if (messagingSystem != null) {
             applyMessagingClientSpan(attributes, telemetry, messagingSystem, spanKind);
             return;
         }
     }
 
-    private static final AttributeKey<String> LOGGER_LEVEL = AttributeKey.stringKey("level");
-    private static final AttributeKey<String> LOGGER_LOGGER_NAME = AttributeKey.stringKey("loggerName");
-    private static final AttributeKey<String> LOGGER_ERROR_STACK = AttributeKey.stringKey("error.stack");
-
-    private void exportLogSpan(SpanData span, Map<AttributeKey<?>, Object> attributes) {
+    private void exportLogSpan(SpanData span) {
         String message = span.getName();
-        String level = removeAttributeString(attributes, LOGGER_LEVEL);
-        String loggerName = removeAttributeString(attributes, LOGGER_LOGGER_NAME);
-        String errorStack = removeAttributeString(attributes, LOGGER_ERROR_STACK);
-        Double samplingPercentage = removeAiSamplingPercentage(attributes);
+        ReadableAttributes attributes = span.getAttributes();
+        String level = attributes.get(AI_LOG_LEVEL_KEY);
+        String loggerName = attributes.get(AI_LOGGER_NAME_KEY);
+        String errorStack = attributes.get(AI_LOG_ERROR_STACK_KEY);
+        Double samplingPercentage = attributes.get(AI_SAMPLING_PERCENTAGE_KEY);
         if (errorStack == null) {
             trackTrace(message, span.getStartEpochNanos(), level, loggerName, span.getTraceId(),
```

Comment:
can we simply pass span to trackTrace(Span span)?  it has 8 input parameters.

Reply:
oh yes! that's much better